src/Hng.Application.Test/Features/Roles/CreateRoleCommandShould.cs

```diff
@@ -0,0 +1,114 @@
+﻿using AutoMapper;
+using Hng.Application.Features.Roles.Command;
+using Hng.Application.Features.Roles.Dto;
+using Hng.Application.Features.Roles.Handler;
+using Hng.Domain.Entities;
+using Hng.Infrastructure.Repository.Interface;
+using Moq;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Hng.Application.Test.Features.Roles
+{
+    public class CreateRoleCommandShould
+    {
+        private readonly Mock<IRepository<Domain.Entities.Organization>> _mockOrganizationRepository;
+        private readonly Mock<IRepository<Role>> _mockRoleRepository;
+        private readonly Mock<IMapper> _mockMapper;
```

Comment:
It would be best for the IMapper instance not to be mocked in the tests. Instead, you can use the actual IMapper implementation to ensure that the AutoMapper configuration is correctly validated and functions as expected. 

Using the real IMapper helps to verify that your mappings are properly configured and will catch any issues with your mapping profiles during testing.
### Here is how you can do this:

```
public YourTestClass()
{
    var config = new MapperConfiguration(cfg =>
    {
        cfg.AddProfile<YourMappingProfile>();
    });
    _mapper = config.CreateMapper();
}
```
**Refer to any of the tests for `Jobs `or `Blogs`, or tests written by others.**